src/Client/OAuth2Client.php

```diff
@@ -131,6 +132,7 @@ public function fetchUserFromToken(AccessToken $accessToken)
      */
     public function fetchUser()
     {
+        /** @var AccessToken $token */
```

Comment:
Even with the |AccessToken, the AbstractProvider::getAccessToken that OAuth2Client::getAccessToken calls still only returns an AccessTokenInterface. However, OAuth2Client::fetchUserFromToken as well as the AbstractProvider::getResourceOwner that it calls, take the hard classes. 

In 7.4 the types can be wider/narrower where it makes sense, but in the meantime, we have to suggest to PHPStan that it's it's also the hard class that fetchUserFromToken expects.

Reply:
Yea, that's a mess. I guess checking if it's not an instance of `AccessToken` and throwing an exception is more honest?

Reply:
It could only be one of those two, depending on the version and so if the interface was available (and described only in the docblock return types). The tree of code it runs from that point should only be methods described by the Interface, which is perfectly handled by the AccessToken class anyway. 

So, the comment does for PHPStan, what PHP 7.4 would be able to do for itself - widening the acceptable type.